cli/pkg/local/server.go

```diff
@@ -144,8 +145,63 @@ func (s *Server) DeployValidation(ctx context.Context, r *connect.Request[localv
 		return nil, err
 	}
 	var deployedProjectID string
+	hasProject := false
+	projectIsGithub := false
 	if rc != nil {
 		deployedProjectID = rc.ProjectID
+
+		proj, err := c.GetProjectByID(ctx, &adminv1.GetProjectByIDRequest{
+			Id: deployedProjectID,
+		})
+		if err != nil {
+			// unset if project doesnt exist
+			if errors.Is(err, database.ErrNotFound) {
+				err = dotrillcloud.Delete(s.app.ProjectPath, s.app.adminURL)
+				if err != nil {
+					return nil, err
+				}
+			}
+			deployedProjectID = ""
+		} else {
+			hasProject = true
+			if proj.Project.GithubUrl != "" {
+				projectIsGithub = true
+			}
+		}
+	}
+
+	// get rill user orgs
+	resp, err := c.ListOrganizations(ctx, &adminv1.ListOrganizationsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	userOrgs := make([]string, 0, len(resp.Organizations))
+	for _, org := range resp.Organizations {
+		userOrgs = append(userOrgs, org.Name)
+	}
+	// TODO if len(userOrgs) > 0 then check if any project in these orgs already deploys from ghUrl
+
+	isGithubRepo := true
+	githubRemoteFound := true
+	repoAccess := false
+	var remote *gitutil.Remote
+	var ghURL string
+
+	if !hasProject || projectIsGithub {
+		// check if project is a git repo only if there is no project already, or it is connected to github
+		remote, ghURL, err = gitutil.ExtractGitRemote(s.app.ProjectPath, "", false)
+		if err != nil {
+			if errors.Is(err, git.ErrRepositoryNotExists) {
+				isGithubRepo = false
```

Comment:
We can end up in a state where `isGithubRepo` is false but `githubRemoteFound` is true which doesn't seem correct.

Reply:
Hmm this code was moved around. But I agree it doesnt make sense.